### ISSUE TITLE
ssl_required? error in admin/login path

### ISSUE THREAD

**Issue by samgooi4189** [AlchemyCMS/alchemy-devise#110]
I'm afraid the bug still exists. I have installed Alchemy exactly as instructed and received **undefined method 'ssl_required'** at http://localhost:3000. Then I created a user at http://localhost:3000/admin/signup and everything worked until I logged out. After that I received the following error messages.
http://localhost:3000: **Alchemy::Page not found "/"**
http://localhost:3000/admin/login: **undefined method `ssl_required?**

_Originally posted by @AnniMel in https://github.com/AlchemyCMS/alchemy-devise/issue_comments/679072637_

**Comment by samgooi4189** [comment]
Rails version: 6.0.3.4
Ruby version: ruby 2.6.5p114
alchemy_cms: 5.0.1

already done "rails g alchemy:devise:install"

but still getting errors
`
NoMethodError (undefined method `ssl_required?' for #<Alchemy::Admin::UserSessionsController:0x00007f4ee9352520>
Did you mean?  require):
  
alchemy-devise (4.5.0) app/controllers/alchemy/admin/user_sessions_controller.rb:9:in `block in <class:UserSessionsController>'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:428:in `instance_exec'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:428:in `block in make_lambda'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:181:in `block (2 levels) in halting_and_conditional'
actionpack (6.0.3.4) lib/abstract_controller/callbacks.rb:34:in `block (2 levels) in <module:Callbacks>'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:182:in `block in halting_and_conditional'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:513:in `block in invoke_before'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:513:in `each'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:513:in `invoke_before'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:134:in `run_callbacks'
actionpack (6.0.3.4) lib/abstract_controller/callbacks.rb:41:in `process_action'
actionpack (6.0.3.4) lib/action_controller/metal/rescue.rb:22:in `process_action'
actionpack (6.0.3.4) lib/action_controller/metal/instrumentation.rb:33:in `block in process_action'
activesupport (6.0.3.4) lib/active_support/notifications.rb:180:in `block in instrument'
activesupport (6.0.3.4) lib/active_support/notifications/instrumenter.rb:24:in `instrument'
activesupport (6.0.3.4) lib/active_support/notifications.rb:180:in `instrument'
actionpack (6.0.3.4) lib/action_controller/metal/instrumentation.rb:32:in `process_action'
actionpack (6.0.3.4) lib/action_controller/metal/params_wrapper.rb:245:in `process_action'
activerecord (6.0.3.4) lib/active_record/railties/controller_runtime.rb:27:in `process_action'
actionpack (6.0.3.4) lib/abstract_controller/base.rb:136:in `process'
actionview (6.0.3.4) lib/action_view/rendering.rb:39:in `process'
actionpack (6.0.3.4) lib/action_controller/metal.rb:190:in `dispatch'
actionpack (6.0.3.4) lib/action_controller/metal.rb:254:in `dispatch'
actionpack (6.0.3.4) lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
actionpack (6.0.3.4) lib/action_dispatch/routing/route_set.rb:33:in `serve'
actionpack (6.0.3.4) lib/action_dispatch/routing/mapper.rb:18:in `block in <class:Constraints>'
actionpack (6.0.3.4) lib/action_dispatch/routing/mapper.rb:48:in `serve'
actionpack (6.0.3.4) lib/action_dispatch/journey/router.rb:49:in `block in serve'
actionpack (6.0.3.4) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.0.3.4) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.0.3.4) lib/action_dispatch/routing/route_set.rb:834:in `call'
railties (6.0.3.4) lib/rails/engine.rb:527:in `call'
railties (6.0.3.4) lib/rails/railtie.rb:190:in `public_send'
railties (6.0.3.4) lib/rails/railtie.rb:190:in `method_missing'
actionpack (6.0.3.4) lib/action_dispatch/routing/mapper.rb:19:in `block in <class:Constraints>'
actionpack (6.0.3.4) lib/action_dispatch/routing/mapper.rb:48:in `serve'
actionpack (6.0.3.4) lib/action_dispatch/journey/router.rb:49:in `block in serve'
actionpack (6.0.3.4) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.0.3.4) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.0.3.4) lib/action_dispatch/routing/route_set.rb:834:in `call'
dragonfly (1.2.1) lib/dragonfly/middleware.rb:14:in `call'
warden (1.2.9) lib/warden/manager.rb:36:in `block in call'
warden (1.2.9) lib/warden/manager.rb:34:in `catch'
warden (1.2.9) lib/warden/manager.rb:34:in `call'
rack (2.2.3) lib/rack/tempfile_reaper.rb:15:in `call'
rack (2.2.3) lib/rack/etag.rb:27:in `call'
rack (2.2.3) lib/rack/conditional_get.rb:27:in `call'
rack (2.2.3) lib/rack/head.rb:12:in `call'
actionpack (6.0.3.4) lib/action_dispatch/http/content_security_policy.rb:18:in `call'
rack (2.2.3) lib/rack/session/abstract/id.rb:266:in `context'
rack (2.2.3) lib/rack/session/abstract/id.rb:260:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/cookies.rb:648:in `call'
activerecord (6.0.3.4) lib/active_record/migration.rb:567:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (6.0.3.4) lib/active_support/callbacks.rb:101:in `run_callbacks'
actionpack (6.0.3.4) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/debug_exceptions.rb:32:in `call'
web-console (4.0.4) lib/web_console/middleware.rb:132:in `call_app'
web-console (4.0.4) lib/web_console/middleware.rb:28:in `block in call'
web-console (4.0.4) lib/web_console/middleware.rb:17:in `catch'
web-console (4.0.4) lib/web_console/middleware.rb:17:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
railties (6.0.3.4) lib/rails/rack/logger.rb:37:in `call_app'
railties (6.0.3.4) lib/rails/rack/logger.rb:26:in `block in call'
activesupport (6.0.3.4) lib/active_support/tagged_logging.rb:80:in `block in tagged'
activesupport (6.0.3.4) lib/active_support/tagged_logging.rb:28:in `tagged'
activesupport (6.0.3.4) lib/active_support/tagged_logging.rb:80:in `tagged'
railties (6.0.3.4) lib/rails/rack/logger.rb:26:in `call'
sprockets-rails (3.2.2) lib/sprockets/rails/quiet_assets.rb:13:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
request_store (1.5.0) lib/request_store/middleware.rb:19:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/request_id.rb:27:in `call'
rack (2.2.3) lib/rack/method_override.rb:24:in `call'
rack (2.2.3) lib/rack/runtime.rb:22:in `call'
activesupport (6.0.3.4) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/executor.rb:14:in `call'
dragonfly (1.2.1) lib/dragonfly/cookie_monster.rb:9:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/static.rb:126:in `call'
rack (2.2.3) lib/rack/sendfile.rb:110:in `call'
actionpack (6.0.3.4) lib/action_dispatch/middleware/host_authorization.rb:82:in `call'
webpacker (4.3.0) lib/webpacker/dev_server_proxy.rb:23:in `perform_request'
rack-proxy (0.6.5) lib/rack/proxy.rb:57:in `call'
railties (6.0.3.4) lib/rails/engine.rb:527:in `call'
puma (4.3.6) lib/puma/configuration.rb:228:in `call'
puma (4.3.6) lib/puma/server.rb:713:in `handle_request'
puma (4.3.6) lib/puma/server.rb:472:in `process_client'
puma (4.3.6) lib/puma/server.rb:328:in `block in run'
puma (4.3.6) lib/puma/thread_pool.rb:134:in `block in spawn_thread'
`

**Comment by tvdeyen** [comment]
Which version of `alchemy-devise` are you using? Currently only the master version is working with Alchemy 5. I will release a new version 5.0 soon.

Meanwhile please use

```rb
# Gemfile
...
gem "alchemy-devise", github: "AlchemyCMS/alchemy-devise", branch: "master"
...
```

and run 

```
$ bundle install
```

This should work for now.

**Comment by samgooi4189** [comment]
can I know where is the method 'ssl_required?' reference from?

I checked 
https://github.com/AlchemyCMS/alchemy-devise/blob/master/app/controllers/alchemy/admin/user_sessions_controller.rb
and also
https://github.com/heartcombo/devise/blob/master/app/controllers/devise/sessions_controller.rb

but could not find the method 'ssl_required?'

**Comment by samgooi4189** [comment]
> Which version of `alchemy-devise` are you using? Currently only the master version is working with Alchemy 5. I will release a new version 5.0 soon.
> 
> Meanwhile please use
> 
> ```ruby
> # Gemfile
> ...
> gem "alchemy-devise", github: "AlchemyCMS/alchemy-devise", branch: "master"
> ...
> ```
> 
> and run
> 
> ```
> $ bundle install
> ```
> 
> This should work for now.

I am using alchemy_cms 5.0-stable branch

**Comment by samgooi4189** [comment]
After I changed to alchemy-devise master, it is working.

`gem "alchemy-devise", "~> 4.5"`
is generated by 
`bundle add alchemy-devise`

My bundler version is 2.1.4

**Comment by tvdeyen** [comment]
> After I changed to alchemy-devise master, it is working.

Glad to hear. 

> `gem "alchemy-devise", "~> 4.5"`
> is generated by
> `bundle add alchemy-devise`
> 
> My bundler version is 2.1.4

Bundler only adds Rubygems sources and not GitHub sources.

**Comment by samgooi4189** [comment]
ok, the README of alchemy_cms needs to update https://github.com/AlchemyCMS/alchemy_cms

`
Use Alchemy user

If you don't have your own user class, you can use the Alchemy user model. Just add the following gem into your Gemfile:

$ bundle add alchemy-devise
`

**Comment by tvdeyen** [comment]
No, I do not think we need to update the readme. This is still the correct way to install the gem. The current problem is that we did not released a version to Rubygems that supports Alchemy 5. After we released that version everything is fine again.

**Comment by tvdeyen** [comment]
Released 4.6.0 that includes the fix for Alchemy 5. You can now change back to use the Rubygems version again. It might take a while until it is distributed on all Rubygems CDN, so please give it a couple of minutes.
